buoy/commands/check.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"knative.dev/test-infra/pkg/git"
+	"knative.dev/test-infra/pkg/gomod"
+)
+
+func addCheckCmd(root *cobra.Command) {
+	var domain string
+	var release string
+	var rulesetFlag string
+	var ruleset git.RulesetType
+	var verbose bool
+
+	var cmd = &cobra.Command{
+		Use:   "check go.mod",
+		Short: "Determine if this module has a ref for each dependency for a given release based on a ruleset.",
+		Long: `
+The check command is used to evaluate if each dependency for the given module
+meets the requirements for cutting a release branch. If the requirements are
+met based on the ruleset selected, the command will exit with code 0, otherwise
+an error message is generated and the with the failed dependencies and exit
+code 1. Errors are written to stderr. Verbose output is written to stdout.
+
+Rulesets,
+  Release          check requires all dependencies to have tagged releases.
+  Branch           check requires all dependencies to have a release branch.
+  ReleaseOrBranch  check will use rule (Release || Branch).
+
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// Validation
+			ruleset = git.Ruleset(rulesetFlag)
+			if ruleset == git.InvalidRule {
+				return fmt.Errorf("invalid ruleset, please select one of: [%s]", strings.Join(git.Rulesets(), ", "))
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			gomodFile := args[0]
+
+			var out io.Writer
+			if verbose {
+				out = os.Stdout
+			}
+
+			err := gomod.Check(gomodFile, release, domain, ruleset, out)
+			if errors.Is(err, gomod.DependencyErr) {
+				_, _ = fmt.Fprintln(os.Stderr, err.Error())
```

Comment:
```suggestion
				fmt.Fprintln(os.Stderr, err.Error())
```
do you really need those?

Reply:
if you don't you get warnings